### ISSUE TITLE
Kubernetes connections using exec provided plugin commands fail if client ssl key and cert are not base64 encoded

### ISSUE THREAD

**Issue by tlbrown12** [kr8s-org/kr8s#323]
### Which project are you reporting a bug for?

kr8s

### What happened?

In my kubeconfig I use a third party exec plugin to retrieve certificate data for k8s authentication. The Kr8s authentication logic seems to assume that this exec command always returns data in base64 format which is not the case in my situation. It attempts to decode a string that is not base64 encoded and fails when attempting to make a connection.

```python
import kr8s

client = kr8s.api()

```

```pytb
File "/opt/homebrew/lib/python3.11/site-packages/kr8s/asyncio/_api.py", line 68, in _f
return await _cls(**kwargs, bypass_factory=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/homebrew/lib/python3.11/site-packages/kr8s/_api.py", line 65, in f
await self.auth
File "/opt/homebrew/lib/python3.11/site-packages/kr8s/_auth.py", line 48, in f
await self.reauthenticate()
File "/opt/homebrew/lib/python3.11/site-packages/kr8s/_auth.py", line 60, in reauthenticate
await self._load_kubeconfig()
File "/opt/homebrew/lib/python3.11/site-packages/kr8s/_auth.py", line 181, in _load_kubeconfig
base64.b64decode(self._user["client-certificate-data"])
File "/opt/homebrew/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/base64.py", line 88, in b64decode
return binascii.a2b_base64(s, strict_mode=validate)
```

The authentication logic should check to see if the cert data is base64 encoded before it attempts to decode it.

Create a kube config file with a config that uses an exec command to retrieving credentials. The exec command should return a ExecCredential object that includes ssl key and cert data that is not base64 encoded. Initializing a kr8s client for this command should return this error.

### Anything else?

This happens for both clientCertificateData and clientKeyData values in the returned string. Modifying auth.py directly to remove base64 decoding for the two resolves the issue.

**Comment by jacobtomlinson** [comment]
I've opened #324 with a potential fix for this. Would you mind trying that out and confirming if it resolves your issue @tlbrown12?

**Comment by tlbrown12** [comment]
Hey @jacobtomlinson. Thank you so much for addressing this. So I had to do two things to get this working with the updated logic. 

`        if "client-certificate-data" in self._user:
            async with NamedTemporaryFile(delete=False) as cert_file:
                try:
                    cert_data = base64.b64decode(self._user["client-certificate-data"], validate=True)
                except binascii.Error:
                    cert_data = self._user["client-certificate-data"].encode()
                await cert_file.write_bytes(cert_data)
                self.client_cert_file = str(cert_file)`

I added the validate=True flag to each of the base64.b64decode call. Without it the client-certificate-data specifically would sometimes decode successfully and ultimately throw an SSL error like the one below when it attempts to make a connection to kubernetes. The validate flag which defaults to false will remove what it deems to be invalid characters before it attempts to decode, and it seems that this was causing the issue. 
`    sslcontext.load_cert_chain(
ssl.SSLError: [SSL] PEM lib (_ssl.c:4044)`  

I also added an encode() call to the end of the string on exception. This is because cert_file.write_bytes expects bytes when that object is a string at that time. One I made these two changes it worked like a dream.
